src/components/ListItem.jsx

```diff
@@ -1,28 +1,44 @@
 import './ListItem.css';
 import { updateItem, deleteItem } from '../api/firebase.js';
-import { useState, useEffect, useMemo } from 'react';
+import { useState, useEffect } from 'react';
 import { ONE_DAY_IN_MILLISECONDS } from '../utils/dates.js';
 import { purchaseUrgency } from '../utils/hooks.js';
 
 export function ListItem({ listPath, item }) {
-	/* Returns a boolean that is passed into isChecked useState
-	On render, box is checked if purchased less than a day ago */
+	//Time variables for each item on render
+	const lastPurchase =
+		item.dateLastPurchased[item.dateLastPurchased.length - 1];
+	const timeElapsed = Date.now() - lastPurchase.seconds * 1000;
 
-	const purchasedOneDayAgo = useMemo(() => {
-		if (item.dateLastPurchased === null) {
+	//Boolean to pass into isChecked state
+	const purchasedWithinDay = () => {
+		//Exclude new, never purchased items - new items are added to db with same dateCreated + first dateLastPurchased
+		if (item.dateCreated.seconds === lastPurchase.seconds) {
 			return false;
+		} else {
+			return timeElapsed <= ONE_DAY_IN_MILLISECONDS;
 		}
+	};
+
+	//Box is checked on render if purchased within 24 hrs
+	const [isChecked, setIsChecked] = useState(purchasedWithinDay);
 
-		const timeDiff = Date.now() - item.dateLastPurchased.seconds * 1000;
-		return timeDiff <= ONE_DAY_IN_MILLISECONDS;
-	}, [item.dateLastPurchased]);
+	//If item is checked on render, calculate time until isChecked state is set to false/unchecked
+	useEffect(() => {
+		if (isChecked) {
+			const timeRemaining = ONE_DAY_IN_MILLISECONDS - timeElapsed;
+			const timer = setTimeout(() => {
+				setIsChecked(false);
+			}, timeRemaining);
+			return () => clearTimeout(timer);
+		}
+	}, [isChecked, timeElapsed]);
 
-	const [isChecked, setIsChecked] = useState(purchasedOneDayAgo);
 	const changeHandler = (e) => {
-		setIsChecked(!isChecked);
 		async function purchaseItem() {
 			try {
 				await updateItem(listPath, item.id, !isChecked);
+				setIsChecked(!isChecked);
```

Comment:
issue (non-blocking): This is non-blocking for your particular issue but I noticed (I think we had flagged this at some point before too) that if you actively uncheck an item that you checked by accident, the dateNextPurchased doesn't revert back to what it was before. I can add a separate issue for that!